pajbot/bot.py

```diff
@@ -851,17 +852,35 @@ def safe_say(self, message: str, channel: Optional[str] = None) -> None:
     def me(self, message: str, channel: Optional[str] = None) -> None:
         self.say("/me " + message[: CHARACTER_LIMIT - 4], channel=channel)
 
-    def announce(self, message: str, channel: Optional[str] = None) -> None:
+    def announce(self, message: str, channel_id: Optional[str] = None, color: Optional[str] = None) -> None:
         if message is None:
             log.warning("message=None passed to Bot::announce()")
             return
 
         if self.silent:
             return
 
+        if channel_id is None:
+            channel_id = self.streamer.id
+
         message = utils.clean_up_message(message)
 
-        self.privmsg("/announce " + message[: CHARACTER_LIMIT - 10], channel=channel)
+        try:
+            self.twitch_helix_api.send_chat_announcement(
+                channel_id,
+                self.bot_user.id,
+                {"message": message[: CHARACTER_LIMIT - 10], "color": color},
```

Comment:
I'd still like to not have to construct the POST body here, this is a detail that must be abstracted away by the apiwrapper (that's what the apiwrapper is for)